java/java.editor.base/src/org/netbeans/modules/java/editor/base/semantic/SemanticHighlighterBase.java

```diff
@@ -727,6 +630,76 @@ private void handlePossibleIdentifier(TreePath expr, Collection<UseTypes> type,
             }
             
             if (c != null) {
+                Collection<UseTypes> type = EnumSet.noneOf(UseTypes.class);
+
+                if (decl.getKind().isClass() || decl.getKind().isInterface()) {
+                    if (!declaration) {
+                        type.add(UseTypes.CLASS_USE);
+                    }
+                } else if (decl.getKind().isField() || isLocalVariableClosure(decl)) {
+                    if (!declaration) {
+                        TreePath pathToCheck = getCurrentPath();
+                        while (true) {
+                            if (pathToCheck.getParentPath().getLeaf().getKind() == Kind.POSTFIX_DECREMENT ||
+                                pathToCheck.getParentPath().getLeaf().getKind() == Kind.POSTFIX_INCREMENT ||
+                                pathToCheck.getParentPath().getLeaf().getKind() == Kind.PREFIX_DECREMENT ||
+                                pathToCheck.getParentPath().getLeaf().getKind() == Kind.PREFIX_INCREMENT) {
+                                type.add(UseTypes.WRITE);
+                                pathToCheck = pathToCheck.getParentPath();
+                                continue;
+                            }
+                            if (CompoundAssignmentTree.class.isAssignableFrom(pathToCheck.getParentPath().getLeaf().getKind().asInterface()) &&
+                                ((CompoundAssignmentTree) pathToCheck.getParentPath().getLeaf()).getVariable() == pathToCheck.getLeaf()) {
+                                type.add(UseTypes.WRITE);
+                                pathToCheck = pathToCheck.getParentPath();
+                                continue;
+                            }
+                            break;
+                        }
+                        if (pathToCheck.getParentPath().getLeaf().getKind() == Kind.ASSIGNMENT &&
+                            ((AssignmentTree) pathToCheck.getParentPath().getLeaf()).getVariable() == pathToCheck.getLeaf()) {
+                            type.add(UseTypes.WRITE);
+                        } else if (pathToCheck.getParentPath().getLeaf().getKind() != Kind.EXPRESSION_STATEMENT) {
+                            type.add(UseTypes.READ);
```

Comment:
This is great, even in the case of `idA = idB++`, the idB's use types will include READ.